app/models/shipping.rb

```diff
@@ -0,0 +1,76 @@
+  require 'active_shipping'
+
+  # returns shipping rates and determines the size of a package given weight
+  # uses active shipping to calculate and return rates
+  class Shipping
+
+    SMALL = [5, 5, 5]
+    MEDIUM = [10, 10, 10]
+    LARGE = [20, 20, 20]
+
+
+    # UPS_KEY = "9D0CA9FE6A4849E8"
+
+    def initializer
+
+      # @ups = ActiveShipping::UPS.new(login: ENV["UPS_LOGIN"], password: ENV["UPS_PASSWORD"], key: ENV["UPS_KEY"])
+      @ups = ActiveShipping::UPS.new(login: ENV["UPS_LOGIN"], password: ENV["UPS_PASSWORD"], key: ENV["UPS_KEY"])
+
+    end
+
+    def self.shipping_rates_calculator(origins, destination)
+
+      destination = active_shipping_location(destination)
+
+      total_rates = origins.map do |origin_hash|
+
+        origin = active_shipping_location(origin_hash["origin"])
+
+        package = active_shipping_package(origin_hash)
+        @ups = ActiveShipping::UPS.new(login: ENV["UPS_LOGIN"], password: ENV["UPS_PASSWORD"], key: ENV["UPS_KEY"])
+
```

Comment:
It looks like you tried to set up @ups as an instance variable above, and I think it you had used initialize instead of initializer it would have worked so you could just use it here instead of defining again. Although it looks like you only use this variable here within this method so you could just define it here, and use a local variable ups instead of instance variable @ups.